pyvrp/Population.py

```diff
@@ -2,18 +2,49 @@
 
 from bisect import insort_left
 from dataclasses import dataclass
-from typing import Callable, Iterator, List, NamedTuple, Tuple
+from typing import (
+    Callable,
+    Generic,
+    Iterator,
+    List,
+    NamedTuple,
+    Optional,
+    Protocol,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from pyvrp.diversity.BiasedFitness import BiasedFitness
 
 from .Individual import Individual
 from .PenaltyManager import PenaltyManager
 from .ProblemData import ProblemData
 from .XorShift128 import XorShift128
 
-_DiversityMeasure = Callable[[ProblemData, Individual, Individual], float]
+
+class PopulationIndividual(Protocol):  # pragma: no cover
+    """
+    Defines an individual that can be put in a population target protocol.
+    """
+
+    def cost(self) -> Union[int, float]:
+        ...
+
+    def is_feasible(self) -> bool:
+        ...
+
+
+TIndiv = TypeVar("TIndiv", bound=PopulationIndividual)
```

Comment:
Maybe `Solution`? Or `SolutionType`? `TIndiv` does not immediately ring a bell for me.